### ISSUE TITLE
Crash if composite not avialable

### ISSUE THREAD

**Issue by lycsjm** [nine7nine/Easystoke-n7n-pkgbuild#2]
Thanks for update this project.
I tried it on my Arch, it can build successfully, but it crashed with following message:

```
** (easystroke:443397): WARNING **: 00:43:55.894: Error: Couldn't read preferences.

** (easystroke:443397): WARNING **: 00:43:55.895: Loaded actions.

** Message: 00:43:55.900: Opened Device 12 ('DEFT Pro TrackBall').

** (easystroke:443397): ERROR **: 00:43:55.900:
unhandled exception (type std::exception) in signal handler:
what: 'composite' not available

zsh: trace trap (core dumped)  easystroke
```

I am using i3-wm without composite.

**Comment by nine7nine** [comment]
hey, sorry for the delayed response.

It looks like when i picked up the commit to remove compiz support, it also hosed non-composited support. 

look here: dcf02288c64a993f647e27ccd186f96df5c9c0e8

we could see if reverting that change makes it work for you. Then I could look at adapting that commit to keep
non-composited setups. ~ my only issue is that I won't be testing non-composite. I only use Gnome/Muttter on X 

I'll try and publish a pkgbuild for you to try later ~ or you can try reverting that commit yourself and get back to me.

**Comment by nine7nine** [comment]
@lycsjm [lycsjm](https://github.com/lycsjm)

can you please try building and testing the latest code?

I've restored the Xshape support and composite related tracing. https://github.com/nine7nine/Easystroke-n7n/commit/917c512f23a2cc8c91edcfa40fd8b3698d048f31

it should work for you now. but I haven't tested without composite (and won't be). so let me know, if it fixes the crash.

you may need to set "Method to Show Gestures" to XShape in preferences. not sure.

**Comment by nine7nine** [comment]
also, any future issues can be reported in my easystroke source code repo, rather than here. ~ i've enabled issues

**Comment by lycsjm** [comment]
@nine7nine 
thanks for update, I build the lastest code and now it works well.
Also thanks for open issue in source code.

**Comment by nine7nine** [comment]
No problem. I'm glad that we have it working now. Thanks for pointing out the issue.

tbh, I hadn't review that compiz commit enough - as I wouldn't have removed the non-composited code, had I known. I'm just familiarizing myself with Easystroke's codebase and I'm not really a C++ guy... it's still coming along though.

I'll just need to be a little more careful, when pulling from other forks. review better.

**Comment by lycsjm** [comment]
I'm not familiar with C++ or even PKGBUILD.
After your first reply, I have tried revert that commit , but got stuck on how to apply change into PKGBUILD .

Before Original repo in AUR can't build few weeks ago, and no other folks can work normally, 

A few weeks before I found this repo, the original ones failed to compile and no other folks were available. Alternatives to easystroke are also lacking on Linux, so even if you don't specialize in C++, thanks for providing available update.

**Comment by nine7nine** [comment]
Yeah, I had just re-installed Arch after an m.2 SSD upgrade - and Easystroke wouldn't build. it was irritating, so I did some digging around and had it building again, but I then decided to fork my own branch; as I noticed there were lots of patches and forks floating around.

I'm familiar enough with C++, but I tend to stick more to C. Anyway, you aren't the only one affected by Easystoke in AUR being broken... and ultimately, I'm not sure the current maintainer is overly committed to it. Being as my laptop has a Stylus and I've been using Easystroke for about 10 years or so, I'll probably be staying on top of maintaining my fork -- I may not be quick to implement new/visible features, but I will try to fix things and keep it working, as expected.

no problem at all on providing packages and updates (I used to be a maintainer for packages in the AUR, once upon a time.)
